profiles/api.py

```diff
@@ -0,0 +1,30 @@
+"""
+Functions for profiles
+"""
+import logging
+
+from django.core.exceptions import ObjectDoesNotExist
+
+from backends.edxorg import EdxOrgOAuth2
+
+log = logging.getLogger(__name__)
+
+
+def get_social_username(user):
+    """
+    Get social auth edX username for a user, or else return None.
+
+    Args:
+        user (django.contrib.auth.models.User):
+            A Django user
+    """
+    if user.is_anonymous():
+        return None
+
+    try:
+        return user.social_auth.get(provider=EdxOrgOAuth2.name).uid
+    except ObjectDoesNotExist:
+        return None
+    except Exception as ex:  # pylint: disable=broad-except
+        log.error("Unexpected error retrieving social auth username: %s", ex)
+        return None
```

Comment:
We generally don't test logging, but it seems like we should here since we have no information about the problem otherwise. I'll add this to the test